src/AmplifyMapLibreRequest.ts

```diff
@@ -89,12 +87,15 @@ export default class AmplifyMapLibreRequest {
 
       // Refresh credentials on a timer because HubEvents do not trigger on credential refresh currently
       this.activeTimeout && clearTimeout(this.activeTimeout);
-      const expiration = new Date(this.credentials.expiration);
-      const timeout = expiration.getTime() - new Date().getTime() - 10000; // Adds a 10 second buffer time before the next refresh
-      this.activeTimeout = window.setTimeout(
-        this.refreshCredentialsWithRetry,
-        timeout
-      );
+      // Refresh credentials when expiration time is later than now
+      if (this.credentials.expiration.getTime() > (new Date()).getTime()) {
```

Comment:
`refreshCredentialsWithRetry` is the handler for both `signOut` and `refreshToken` Hub events, and since these are async, we could still have a race condition scenario, of `refreshToken` being handled at same time as `signOut` but `signOut` resolves faster and an error is thrown on refreshToken retry. No infinite loop but we could still see an error, correct me if I am wrong. My question here is why can't we avoid getting credentials on signOut?

Reply:
@ioanabrooks We can avoid getting credentials on signOut certainly but the original reason for the change was that if a user was showing a map and went from signed in user to a guest user, they should no longer have access to call the map API. I've tested this in an example app recently and after signing out I could no longer make further requests to get map tiles (I could still see the parts of the map I loaded though).

Also I believe the refreshToken event is only fired when we attempt to get new identity token and access token. As far as I'm aware that is not the same scenario as a signout or signin event. Though there is a case where a user signsOut then instantly attempts a refreshToken. In that case we would throw an error on the attempted refreshToken as we will also try to refreshCredentials.

Reply:
Got it, so maybe we can mitigate this by handling `tokenRefresh_failure` hub event. what do you think?

Reply:
Yep we would just need to verify that, I can see I even left a comment that says that refreshToken is not currently fired - https://github.com/aws-amplify/maplibre-gl-js-amplify/pull/194/files#diff-3962f2f1d3b5c5b6449fb8fbed0fe5a2a00142d04fb3449107d130ea9f577aa9R88

Reply:
I'd still like to see the `tokenRefresh_failure` event handled, and if the token refresh event does not get dispatched as expected, we should include a comment referencing reference why that is for future maintainers, open a github issue if one does not exist because users can customize the refresh rate to lower or greater than an hour on Cognito side.